File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/03_Medium/10324_:::_Zeros_and_Ones/Main.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main
{
	public static void main(String[] args) throws IOException
	{
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		//-----------------------------------------------------------
		int c=1, casos, a, b, init, fin;
		boolean iguales;
		char anterior, leido[];
		for(String linea, aux[]; (linea=br.readLine())!=null;)
		{
			System.out.println("Case "+(c++)+":");
			casos=Integer.parseInt(br.readLine());
			leido=linea.toCharArray();

			while(casos--!=0)
			{
				aux=br.readLine().split(" ");
				a=Integer.parseInt(aux[0]);
				b=Integer.parseInt(aux[1]);
				init=Math.min(a, b);
				fin=Math.max(a, b);

				anterior=leido[init];
				iguales=true;
				for(int i=init+1; i<=fin && iguales; i++)
				{
					if(anterior!=leido[i]) iguales=false;
					anterior=leido[i];
				}
				System.out.println(iguales?"Yes":"No");
			}
		}
		//-----------------------------------------------------------
		br.close();
	}
}

File: Capitulo_01/01_Getting_Started:_The_Easy_Problems/03_Medium/10324_:::_Zeros_and_Ones/Main.cpp
using namespace std;
#include <cstdio>
#include <algorithm>

int main()
{
	char linea[1000005];
	int c=1, casos, a, b, init, fin;
	char anterior;
	bool iguales;

	while(scanf("%s", linea)!=EOF)
	{
		printf("Case %d:\n", c++);

		scanf("%d", &casos);
		while(casos--)
		{
			scanf("%d %d", &a, &b);
			init=min(a,b);
			fin=max(a,b);

			anterior=linea[init];
			iguales=true;
			for(int j=init+1; j<=fin; j++)
			{
				if(anterior!=linea[j])
				{
					iguales=false;
					break;
				}
				anterior=linea[j];
			}
			puts(iguales?"Yes":"No");
		}
	}
	return 0;
}
